torch/csrc/distributed/c10d/TCPStoreLibUvBackend.cpp

```diff
@@ -713,70 +713,102 @@ class UvClient : public UvTcpSocket {
       if (!stream.read1(command))
         break;
       if (store->isMiscellaneousClient(iptr())) {
-        if ((QueryType)command != QueryType::VALIDATE)
+        if ((QueryType)command != QueryType::VALIDATE) {
+          stream.commit();
```

Comment:
Yikes, this seems error prone in the future. Wonder if this is a where to refactor, potentially with a RAII logic?

Reply:
ACK! Will make a new PR.